File: LyBaiduMapProtocol/LyBaiduMapProtocol/地图实现/百度/LyBaiduMapFactory.m
//
//  LyBaiduMapFactory.m
//  LyBaiduMapProtocol
//
//  Created by 张杰 on 2017/6/19.
//  Copyright © 2017年 张杰. All rights reserved.
//

#import "LyBaiduMapFactory.h"
#import <BaiduMapAPI_Map/BMKMapView.h>//只引入所需的单个头文件
#import "LyBaiduMapView.h"
#import "LyBaiduMapLocal.h"

@interface LyBaiduMapFactory ()


//@property(nonatomic,strong)
@property(nonatomic,strong)BMKMapManager *mapManager;

@end

@implementation LyBaiduMapFactory

//1.配置地图key
- (void)setupAppKey:(NSString *)appKey
{
    self.mapManager = [[BMKMapManager alloc]init];
    [self.mapManager start:appKey generalDelegate:nil];
}

//2.展示地图map
- (id<LyMapViewProtocol>)loadMapWithFrame:(CGRect)frame
{
    LyBaiduMapView *mapView = [[LyBaiduMapView alloc] initWithFrame:frame];
    
    return mapView;
}

- (id<LyLocalProtocol>)loaclTool
{
    LyBaiduMapLocal *loacl = [[LyBaiduMapLocal alloc] init];
    return loacl;
}

@end

File: LyMapProtocol/LyMapProtocol/百度/工厂/LyBaiduMapFactory.swift
//
//  LyMapFactory.swift
//  LyMapProtocol
//
//  Created by 张杰 on 2017/6/19.
//  Copyright © 2017年 张杰. All rights reserved.
//

import UIKit


class LyBaiduMapFactory: NSObject, LyMapFactoryProtocol {
    
    required init(_ appKey: String) {
        super.init()
        if !isSetting {
            isSetting = true
            manager.start(appKey, generalDelegate: nil)
        }
    }
    
    func loadMapView(_ frame : CGRect) -> LyMapViewProtocol {
        
        return LyBaiduMapView(frame)
    }
    
    fileprivate let manager = BMKMapManager()
    fileprivate var isSetting = false
}
